hw1/task05/task05.py

```diff
@@ -0,0 +1,29 @@
+"""
+Given a list of integers numbers "nums".
+
+You need to find a sub-array with length less equal to "k", with maximal sum.
+The written function should return the sum of this sub-array.
+Examples:
+    nums = [1, 3, -1, -3, 5, 3, 6, 7], k = 3
+    result = 16
+"""
+
+
+def find_maximal_subarray_sum(nums: list, k: int) -> int:
+    """Max sum of continuous sub-array."""
+    ans = -float("INF")
+    index = 0
+    n = len(nums)
+    # fixing situation where k > N
+    k = min(n, k)
+    # find max subarray sum and its start index
+    for i in range(n - k + 1):
+        if sum(nums[i : i + k]) > ans:
+            ans = sum(nums[i : i + k])
+            index = i
+
+    # exclude negative numbers
+    for number in nums[index : index + k]:
```

Comment:
`pytest.mark.parametrize` is a good thing to remove boilerplate, but it should be used with caution. 

You have grouped tests that cover different cases.   I'd recommend writing a separate for that particular test case.   

I have some doubts about this code.  How can we be sure that this is still max value, after reducing it?

Reply:
We are checking if a number is negative. So double negation do increase max sum value.

Reply:
About tests, what do you have in mind?